sstables/kl/reader_impl.hh

```diff
@@ -159,8 +158,9 @@ private:
     temporary_buffer<char>* _processing_data;
 public:
     using consumer = row_consumer;
+    // assumes !primitive_consumer::active()
     bool non_consuming() const {
-        return !_consuming;
+        return false;
```

Comment:
‎This _consuming thing was added in a previous patch, but now removed. So why add it at all?

Reply:
Only after introducing the _consuming variable I realized that we only check non_consuming if !primitive_consumer::active(), and only then I realized that there are no states in kl reader's non_consuming() OR, where !primitive_consumer::active()